client/fingerprint/fingerprint.go

```diff
@@ -3,14 +3,27 @@ package fingerprint
 import (
 	"fmt"
 	"log"
+	"sort"
 	"time"
 
 	"github.com/hashicorp/nomad/client/config"
 	"github.com/hashicorp/nomad/nomad/structs"
 )
 
 // EmptyDuration is to be used by fingerprinters that are not periodic.
-const EmptyDuration = time.Duration(0)
+const (
+	EmptyDuration = time.Duration(0)
+)
+
+func init() {
+	// Initialize the list of available fingerprinters per platform.  Each
+	// platform defines its own list of available fingerprinters.
+	initFingerprints(builtinFingerprintMap)
```

Comment:
Do you think it would be better to have the common ones here and add a method called `platformSpecificFingerprinters()` which we can implement for different platforms for adding the ones which are platform specific? Most of the fingerprinters are common besides the resource isolation ones.